### PR TITLE
Make forever and pause be top level

### PR DESCRIPTION
**Pull request by riknoll**
Moving both of those functions to be top-level, just like `pauseUntil()`. In other words:
1. `loops.forever(() => {})` is now `forever(() => {})`
2. `loops.pause(500)` is now `pause(500)`

This shouldn't break anybody. The new functions I defined have the same block id as the old functions so hopefully old block scripts will be automatically converted when we compile to TS.

The old functions are still there so that we don't break TS scripts, but they have `deprecated=true` and I changed the block ids.

Adding DO NOT MERGE because I want to make sure everyone is aware of this change before I merge it in.

### CODE REVIEW

**Comment by mmoskal** [comment]
I can't even tell you how glad I am to see `pause` leave `loops`. It always twisted my mind in weird ways...

File: libs/base/control.ts [hunk 1]
```diff
@@ -178,6 +178,27 @@ function pauseUntil(condition: () => boolean, timeOut?: number): void {
...
+
+/**
+ * Pause for the specified time in milliseconds
+ * @param ms how long to pause for, eg: 100, 200, 500, 1000, 2000
+ */
+//% help=loops/pause weight=99
+//% async block="pause %pause=timePicker|ms"
+//% blockId=device_pause blockNamespace="loops"
+function pause(ms: number): void {
+    loops.pause(ms);
```
Comment (pelikhan) [inline libs/base/control.ts]: shouldn it be the other way around? loops.pause calls into pause?

File: libs/base/loops.cpp [hunk 2]
```diff
@@ -16,9 +16,9 @@ void forever(Action a) {
  * Pause for the specified time in milliseconds
  * @param ms how long to pause for, eg: 100, 200, 500, 1000, 2000
  */
-//% help=loops/pause weight=99
+//% help=loops/pause weight=99 deprecated=true
 //% async block="pause %pause=timePicker|ms"
-//% blockId=device_pause
+//% blockId=device_pause_deprecated
```
Comment (samelhusseini) [inline libs/base/loops.cpp]: If you change the block id, it's no longer backwards compatible..

Comment (samelhusseini) [reply libs/base/loops.cpp]: oh nevermind, you have device_pause elsewhere.

**Comment by samelhusseini** [comment]
Question RE highlighting? I don't see anything setting the highlight colors of that. Most top level blocks use the standard monaco blue (eg: while, for, etc). I would say we highlight those with the same color. You can add that in monaco.tsx:defineEditorTheme

**Comment by riknoll** [comment]
It should probably use the loops color right? Just whatever we set as blockNamespace since that will be the color in the blocks

**Comment by pelikhan** [comment]
Yes loops color to match the category color.

**Comment by samelhusseini** [comment]
I'm on the fence on that. On the one hand you want it to appear as a lang construct, hence why it's top level. 
On the other hand, it's in loops in Blockly, so maybe should match color there.. @others?

**Comment by pelikhan** [comment]
We have a convention that all blocks under a category have the same color, let's keep that. I know that Chibi already has an exception.

**Comment by samelhusseini** [comment]
All good.

**Pull request merged**